packages/massa-web3/src/experimental/basicElements/internal.ts

```diff
@@ -17,11 +17,12 @@ export function extractData(
   return extractedData
 }
 
-export function checkPrefix(str: string, ...expected: string[]): void {
+export function checkPrefix(str: string, ...expected: string[]): string {
```

Comment:
Badly named. If you return the prefix, it should be something like `mustExtractPrefix`. The "must" is there to explicitly indicate that a failure will throw an exception.